src/rust/sqs-executor/src/lib.rs

```diff
@@ -306,16 +334,37 @@ async fn process_message<
                 error=?e,
                 recoverable=?e.error_type()
             );
-            let event = serializer
+            let events = serializer
                 .serialize_completed_events(&[partial])
                 .expect("Serializing failed");
+            // emit event
+            // todo: we should retry event emission
+            if events.is_empty() {
+                tracing::debug!(message = "Serialized events produced no output");
+                return;
+            }
+
+            let mut envelopes = vec![];
+            for event in events {
+                let envelope = rust_proto::services::Envelope {
+                    metadata: Some(meta.clone()),
+                    inner_type: "".to_string(),
```

Comment:
Is this intentional? The `inner_type` value should be the fully-qualified package of the protobuf type which was used to serialize `inner_message`.

Reply:
Not intentional, thanks for catching - I should have used a todo!() so that the tests failed.